source/stylist/fortran.py

```diff
@@ -295,3 +297,58 @@ def examine_fortran(self, subject: FortranSource) -> List[Issue]:
 
         issues.sort(key=lambda x: (x.filename, x.line, x.description))
         return issues
+
+
+class KindPattern(FortranRule):
+    def __init__(self,
+                 integer_pattern: Union[str, Pattern],
+                 real_pattern: Union[str, Pattern]):
+        self._patterns: Dict[str, Pattern] \
+            = defaultdict(lambda: re.compile(r'.*'))
```

Comment:
Could do with an explanatory comment here perhaps - I _think_ this means that if you try to retrieve a pattern from the dictionary where one was never set the lambda function will be used to return a match object that matches "anything" because in that instance you must not care what matches?

Reply:
Having said that... you only ever set (and _always_ set `'integer'` and `'real'` below) so unsure how the case the `defaultdict` is allowing for would ever come up?  (and if it did, wouldn't you rather have it fail because the tool is trying to request a pattern where one doesn't exist?)

Reply:
Fortran supports lots of other data types, for instance logical and complex. We may care about complex in the future but it really makes no sense to consider the kind of logical.

You are quite right, the default dictionary contains a "match all" pattern for anything which is not explicitly set.

I think the argument on how to handle this is quite marginal. I chose to provide for future proofing, if any new data types are added to the standard this rule will handle them automatically. However just as good would be the alternative approach of explicitly stating all known types and having it complain when new ones are discovered.

I really don't mind which we go with so please feel free to weigh in.